src/path_manager.c

```diff
@@ -762,6 +777,83 @@ static void handle_mptcp_event(struct l_genl_msg *msg, void *user_data)
         };
 }
 
+/**
+ * @brief Verify that MPTCP is configured as needed in the kernel.
+ *
+ * The multipath-tcp.org kernel exposes several @c sysctl variables,
+ * two of which must be configured in a manner suitable for mptcpd:
+ *
+ * @li @c net.mptcp.mptcp_enabled != 0
+ * @li @c net.mptcp.mptcp_path_manager == netlink
+ *
+ * Issue a warning if they exist, and are not set in a manner suitable
+ * for mptcpd.
+ *
+ * @note Mptcpd should not set these variables since it would need to
+ *       be run as @c root in order to gain write access to files in
+ *       @c /proc/sys/net/mptcp/ owned by @c root.  Ideally, mptcpd
+ *       should not be run as the @c root user.
+ */
+static void check_kernel_config()
+{
+        static char const mptcp_enabled[] =
+                MPTCP_SYSCTL_BASE "mptcp_enabled";
+        static char const mptcp_path_manager[] =
+                MPTCP_SYSCTL_BASE "mptcp_path_manager";
+
+        FILE *f = fopen(mptcp_enabled, "r");
+
+        if (f != NULL) {
+                int enabled = 0;
+                int const n = fscanf(f, "%d", &enabled);
+
+                if (likely(n == 1)) {
+                        // "enabled" could be 0, 1, or 2.
+                        if (enabled == 0) {
+                                l_warn("MPTCP is not enabled in "
+                                       "the kernel.");
+                                l_warn("Try 'sysctl -w "
+                                       "net.mptcp.mptcp_enabled=2'.");
+                        }
+                } else {
+                        l_error("Unable to determine if MPTCP "
+                                "is enabled.");
+                }
+
+                fclose(f);
+        }
+
```

Comment:
It might be good to split this function in two, one `check` function for each `sysctl` variable.

Reply:
Good idea!

Reply:
Okay, I'll split the function.